cuML/src/solver/cd.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "ml_utils.h"
+#include <cuda_utils.h>
+#include <linalg/gemm.h>
+#include <linalg/add.h>
+#include <linalg/subtract.h>
+#include <linalg/multiply.h>
+#include <linalg/eltwise.h>
+#include <linalg/unary_op.h>
+#include <linalg/cublas_wrappers.h>
+#include <matrix/math.h>
+#include <matrix/matrix.h>
+#include "glm/preprocess.h"
+#include "shuffle.h"
+#include <functions/penalty.h>
+#include <functions/softThres.h>
+#include <functions/linearReg.h>
+
+namespace ML {
+namespace Solver {
+
+using namespace MLCommon;
```

Comment:
@teju85 has been pretty adamant about not having `using namespace` in the header files. I personally don't mind it- as in the case of my current UMAP PR, I make heavy use of business objects in MLCommon::Sparse that make the UMAP code almost unreadable unless I use the `using` keyword. We might want to discuss at some point for consistency.

Reply:
I'm ok doing such things inside .cu or .cpp files. The issue with header files is that when they get included in others, such 'namespace imports' will also propagate and it can cause symbol conflicts.

Reply:
I have "using namespace" pretty much everywhere in the code. I'll update all of them in a separate PR.